mirdata/mridangam_stroke.py

```diff
@@ -0,0 +1,254 @@
+        force_overwrite (bool):
+            Whether to overwrite the existing downloaded data
+        cleanup (bool):
+            Whether to delete the zip/tar file after extracting.
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        info_message=None,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if the stored dataset is a valid version
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Return track ids
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load Mridangam Stroke dataset
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    data = {}
+    for key in DATA.index.keys():
+        data[key] = Track(key, data_home=data_home)
+    return data
+
+
+def load_stroke_name(audio_path):
+    """Load stroke name of track
+    Args:
+        audio_path (str): Local path where the track is stored.
+    Returns:
+        stroke_name (str): stroke name or type extracted from filename
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+
+    if '.wav' in audio_path:
+        audio_data = audio_path.split('__')[2]
+        stroke_name = audio_data.split('-')[0]
+
+        return stroke_name
+
+
+def load_tonic(audio_path):
+    """Get tonic of track
+    Args:
+        audio_path (str): Local path where the track is stored.
+    Returns:
+        tonic (str): track tonic extracted from filename
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+
+    if '.wav' in audio_path:
+        audio_data = audio_path.split('__')[2]
+        tonic_info = audio_data.split('-')[1]
```

Comment:
similar to stroke, it would be useful to validate against an existing tonic dictionary 
`assert tonic_info is in self.tonic_dictionary `

Reply:
Done for both!! Thanks for the review!